File: crates/fresh-editor/tests/fixtures/syntax_highlighting/hello.hs
-- Haskell syntax highlighting test
module Main where

greet :: String -> String
greet name = "Hello, " ++ name ++ "!"

factorial :: Integer -> Integer
factorial 0 = 1
factorial n = n * factorial (n - 1)

main :: IO ()
main = do
    let message = greet "World"
    putStrLn message
    print (factorial 10)

File: crates/fresh-editor/tests/fixtures/syntax_highlighting/hello.ml
(* OCaml syntax highlighting test *)
let greet name =
  Printf.sprintf "Hello, %s!" name

let factorial n =
  let rec aux acc = function
    | 0 -> acc
    | n -> aux (acc * n) (n - 1)
  in
  aux 1 n

let () =
  let message = greet "World" in
  print_endline message;
  Printf.printf "Factorial 10 = %d\n" (factorial 10)
